src/clj/puppetlabs/services/jruby_pool_manager/jruby_core.clj

```diff
@@ -94,22 +94,28 @@
   uses. JARS_NO_REQUIRE was the legacy way to turn off jar loading but is
   being phased out in favor of JARS_REQUIRE.  As of JRuby 1.7.20, only
   JARS_NO_REQUIRE is honored.  Setting both of those here for forward
-  compatibility."
+  compatibility.
+
+  We also merge an environment-vars map with the config to allow for whitelisted
+  environment variables to be visible to the Ruby code. This map is by default
+  set to {} if the user does not specify it in the configuration file."
   [env :- jruby-schemas/EnvMap
-   gem-home :- schema/Str]
+   gem-home :- schema/Str
+   config :- (schema/pred map?)]
   (let [whitelist ["HOME" "PATH"]
         clean-env (select-keys env whitelist)]
-    (assoc clean-env
-      "GEM_HOME" gem-home
-      "JARS_NO_REQUIRE" "true"
-      "JARS_REQUIRE" "false")))
+    (merge (assoc clean-env
```

Comment:
IMO the way this is currently seems reasonable/the simplest way of doing things, and trying to make it any more complicated would just be confusing. We already require `GEM_HOME` to be specified in the config, so sure, why not just let them override it if they really want to? We don't actually set anything specific for `HOME` and `PATH`, just let them be taken out of the environment variables. I do think it's worthwhile to document this though.